app/models/service_template_transformation_plan_task.rb

```diff
@@ -176,49 +189,64 @@ def conversion_options
     destination_cluster = transformation_destination(source_cluster)
     destination_storage = transformation_destination(source_storage)
 
-    options = {
+    results = {
       :source_disks     => virtv2v_disks.map { |disk| disk[:path] },
       :network_mappings => network_mappings
     }
 
-    options.merge!(send("conversion_options_source_provider_#{source_ems.emstype}_#{source_transport_method}", source_storage))
-    options.merge!(send("conversion_options_destination_provider_#{destination_ems.emstype}", destination_cluster, destination_storage))
+    results.merge!(send("conversion_options_source_provider_#{source_ems.emstype}_#{source_transport_method}", source_storage))
+    results.merge!(send("conversion_options_destination_provider_#{destination_ems.emstype}", destination_cluster, destination_storage))
+  end
 
+  def update_options(opts)
+    with_lock do
```

Comment:
Now this deserves a comment, what concurrent access are you preventing against?  Is it preventing multiple tasks from seeing this while it is being updated and using old data in other calculations?  Which fields in options specifically?